src/neural-network.js

```diff
@@ -391,46 +392,54 @@ export default class NeuralNetwork {
 
     this._verifyIsInitialized(data);
 
-    while (status.iterations < this.trainOpts.iterations && status.error > this.trainOpts.errorThresh && Date.now() < endTime) {
-      this._trainingTick(data, status);
+    const runTrainingTick = () => {
+      if (status.iterations < this.trainOpts.iterations && status.error > this.trainOpts.errorThresh && Date.now() < endTime) {
+        this._trainingTick(data, status);
+        return true;
+      }
+      return false;
+    };
+
+    return {
+      runTrainingTick,
```

Comment:
See this notation is new to me.  I am still getting the hang of the new js features, but this seems pretty cool.